tests/contrib/grpc/test_grpc.py

```diff
@@ -0,0 +1,173 @@
+
+    def test_span_in_error(self):
+        # Create a channel and send one request to the server
+        with grpc.secure_channel('localhost:%d' % (GRPC_PORT), credentials=grpc.ChannelCredentials(None)) as channel:
+            stub = HelloStub(channel)
+            try:
+                stub.SayError(HelloRequest(name="test"))
+            except:
+                pass # excepted to throw
+
+        writer = self._tracer.writer
+        spans = writer.pop()
+        eq_(len(spans), 1)
+        
+        span = spans[0]
+        eq_(span.error, 1)
+        self.assertIsNotNone(span.meta['error.stack'])
+
+    def test_pin_not_activated(self):
+        self._tracer.configure(enabled=False)
+        Pin.override(grpc, tracer=self._tracer)
+        with grpc.insecure_channel('localhost:%d' % (GRPC_PORT)) as channel:
+            stub = HelloStub(channel)
+            stub.SayHello(HelloRequest(name="test"))
+
+        writer = self._tracer.writer
+        spans = writer.pop()
+        eq_(len(spans), 0)
+
+    def test_pin_tags_are_put_in_span(self):
+        Pin.override(grpc, tags={'tag1': 'value1'})
+        with grpc.insecure_channel('localhost:%d' % (GRPC_PORT)) as channel:
+            stub = HelloStub(channel)
+            stub.SayHello(HelloRequest(name="test"))
+
+        writer = self._tracer.writer
+        spans = writer.pop()
+        eq_(len(spans), 1)
+        span = spans[0]
+        eq_(span.meta['tag1'], 'value1')
+
+    def test_pin_can_be_defined_per_channel(self):
+        Pin.override(grpc, service='grpc1')
+        channel1 = grpc.insecure_channel('localhost:%d' % (GRPC_PORT))
+
+        Pin.override(grpc, service='grpc2')
+        channel2 = grpc.insecure_channel('localhost:%d' % (GRPC_PORT))
+
+        stub1 = HelloStub(channel1)
+        stub2 = HelloStub(channel2)
+        stub1.SayHello(HelloRequest(name="test"))
+        stub2.SayHello(HelloRequest(name="test"))
+
+        writer = self._tracer.writer
+        spans = writer.pop()
+        
+        eq_(len(spans), 2)
+        span1 = spans[0]
+        span2 = spans[1]
+        _check_span(span1, 'grpc1')
+        _check_span(span2, 'grpc2')
+
+        channel1.close()
+        channel2.close()
+
+
+def _check_span(span, service='grpc'):
+    eq_(span.name, 'grpc.client')
+    eq_(span.resource, '/Hello/SayHello')
+    eq_(span.service, service)
+    eq_(span.error, 0)
+    eq_(span.span_type, 'grpc')
+    eq_(span.meta['grpc.host'], 'localhost')
+    eq_(span.meta['grpc.port'], '50531')
+
+
+class SendBackDatadogHeaders(object):
+    def SayHello(self, request, context):
+        """Returns all the headers begining by x-datadog with the following format:
+        header1=value1;header2=value2;...
+        It is used to test propagation
+        """
+        metadata = context.invocation_metadata()
+        context.set_code(grpc.StatusCode.OK)
+        return HelloReply(message=str.join(';', (w.key + '='+ w.value for w in metadata if w.key.startswith('x-datadog'))))
```

Comment:
Line too long, and `';'.join()` is more common than `str.join(';')`

```suggestion
        return HelloReply(
            message=';'.join(w.key + '=' + w.value for w in metadata if w.key.startswith('x-datadog')),
        )
```